gdpr/gdpr.go

```diff
@@ -24,45 +21,22 @@ type Permissions interface {
 	// Determines whether or not to send PI information to a bidder, or mask it out.
 	//
 	// If the consent string was nonsensical, the returned error will be an ErrorMalformedConsent.
-	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, weakVendorEnforcement bool, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
+	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
 }
 
-// Versions of the GDPR TCF technical specification.
-const (
-	tcf2SpecVersion uint8 = 2
-)
-
 // NewPermissions gets an instance of the Permissions for use elsewhere in the project.
-func NewPermissions(ctx context.Context, cfg config.GDPR, vendorIDs map[openrtb_ext.BidderName]uint16, client *http.Client) Permissions {
+func NewPermissions(cfg config.GDPR, tcf2Config TCF2ConfigReader, vendorIDs map[openrtb_ext.BidderName]uint16, fetcher VendorListFetcher) Permissions {
 	if !cfg.Enabled {
 		return &AlwaysAllow{}
 	}
 
-	gdprDefaultValue := SignalYes
-	if cfg.DefaultValue == "0" {
-		gdprDefaultValue = SignalNo
-	}
-
-	purposeConfigs := map[consentconstants.Purpose]config.TCF2Purpose{
-		1:  cfg.TCF2.Purpose1,
-		2:  cfg.TCF2.Purpose2,
-		3:  cfg.TCF2.Purpose3,
-		4:  cfg.TCF2.Purpose4,
-		5:  cfg.TCF2.Purpose5,
-		6:  cfg.TCF2.Purpose6,
-		7:  cfg.TCF2.Purpose7,
-		8:  cfg.TCF2.Purpose8,
-		9:  cfg.TCF2.Purpose9,
-		10: cfg.TCF2.Purpose10,
-	}
-
 	permissionsImpl := &permissionsImpl{
```

Comment:
Before we were passing the top-level GDPR config `config.GDPR` into this function and setting it on the GDPR permissions object. Now we pass in `config.GDPR` as well as `tcf2Config`, which implements the `TCF2ConfigReader` interface, and only set the `tcf2Config` on the object while extracting a couple of values from `config.GDPR` to set directly on the object. IMO, having both configs on the object would be confusing. I also figured it would be best to do it this way instead of just setting `config.GDPR` on the object where `config.GDPR` implements the `TCF2ConfigReader` interface as I thought the interface methods made more send on the `config.GDPR.TCF2` object. Thoughts?